tests/samplers_tests/test_nsgaii.py

```diff
@@ -603,17 +605,61 @@ def test_crossover_inlined_categorical_distribution() -> None:
 
 
 @pytest.mark.parametrize(
-    "crossover,expected_params",
+    "crossover",
+    [
+        UniformCrossover(),
+        BLXAlphaCrossover(),
+        SPXCrossover(),
+        SBXCrossover(),
+        VSBXCrossover(),
+        UNDXCrossover(),
+    ],
+)
+def test_crossover_duplicated_param_values(crossover: BaseCrossover) -> None:
+
+    param_values = [1.0, 2.0]
+
+    study = optuna.study.create_study()
+    rng = np.random.RandomState()
+    search_space = {"x": FloatDistribution(1, 10), "y": IntDistribution(1, 10)}
+    numerical_transform = _SearchSpaceTransform(search_space)
+    parent_params = np.array([param_values, param_values])
+
+    if crossover.n_parents == 3:
+        parent_params = np.append(parent_params, [param_values], axis=0)
+
+    child_params = crossover.crossover(parent_params, rng, study, numerical_transform.bounds)
+    assert child_params.ndim == 1
+    np.testing.assert_almost_equal(child_params, param_values)
+
+
+@pytest.mark.parametrize(
+    "crossover,rand_value,expected_params",
     [
-        (UniformCrossover(), np.array([3.0, 4.0])),
-        (BLXAlphaCrossover(), np.array([2.0, 3.0])),
-        (SPXCrossover(), np.array([2.75735931, 3.75735931])),
-        (SBXCrossover(), np.array([3.0, 4.0])),
-        (VSBXCrossover(), np.array([3.0, 4.0])),
-        (UNDXCrossover(), np.array([1.0, 2.0])),
+        (UniformCrossover(), 0.0, np.array([1.0, 2.0])),  # p1.
+        (UniformCrossover(), 0.5, np.array([3.0, 4.0])),  # p2.
+        (UniformCrossover(), 1.0, np.array([3.0, 4.0])),  # p2.
+        (BLXAlphaCrossover(), 0.0, np.array([0.0, 1.0])),  # p1 - [1, 1].
+        (BLXAlphaCrossover(), 0.5, np.array([2.0, 3.0])),  # (p1 + p2) / 2.
+        (BLXAlphaCrossover(), 1.0, np.array([4.0, 5.0])),  # p2 + [1, 1].
+        # G = [3, 4], xks=[[-1, 0], [3, 4]. [7, 8]].
+        (SPXCrossover(), 0.0, np.array([7, 8])),  # rs = [0, 0], xks[-1].
+        (SPXCrossover(), 0.5, np.array([2.75735931, 3.75735931])),  # rs = [0.5, 0.25].
+        (SPXCrossover(), 1.0, np.array([-1.0, 0.0])),  # rs = [1, 1], xks[0].
+        (SBXCrossover(), 0.0, np.array([2.0, 3.0])),  # c1 = (p1 + p2) / 2.
+        (SBXCrossover(), 0.5, np.array([3.0, 4.0])),  # p2.
+        (SBXCrossover(), 1.0, np.array([3.0, 4.0])),  # p2.
+        (VSBXCrossover(), 0.0, np.array([2.0, 3.0])),  # c1 = (p1 + p2) / 2.
+        (VSBXCrossover(), 0.5, np.array([3.0, 4.0])),  # p2.
+        (VSBXCrossover(), 1.0, np.array([3.0, 4.0])),  # p2.
+        (UNDXCrossover(), 0.0, np.array([2.0, 3.0])),  # [2, 3] + [0, 0] + [0, 0].
+        (UNDXCrossover(), 0.5, np.array([1.0, 2.0])),  # [2, 3] + [-1, -1] + [1, 1].
```

Comment:
In this line, the value of `expected_params` and the comment are inconsistent?

Reply:
You are absolutely right. Thank you for pointing it out. I fixed the comment in https://github.com/optuna/optuna/pull/3705/commits/6ef5216b8c20c8c1e4870d2d1b03b689d2c91623.
In fact, third element is always `0` since all parents are on a line (i.e., `x + 1`).

I also added the negative `rand_value`, so that `UNDXCrossover` could generate values from [2, 3] to [3, 4].